### PR TITLE
Removed deprecated numpy.matrix

### LINKED ISSUE #15563: sympy.physics.quantum.matrixutils uses deprecated np.matrix
Numpy is deprecating `np.matrix`. It is used [here](https://github.com/sympy/sympy/blob/sympy-1.3/sympy/physics/quantum/matrixutils.py#L58) in physic/quantum/matrixutils. Running under pytest I see lots of `PendingDeprecationWarnings`:
```
  /Users/enojb/current/sympy/sympy/sympy/physics/quantum/matrixutils.py:58:
PendingDeprecationWarning: the matrix subclass is not the recommended way to
represent matrices or deal with linear algebra
(see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please
adjust your code to use regular ndarray.
```
I'm not sure how this module is used but maybe SymPy should change this or deprecate these functions as well.

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Closes #15563

#### Brief description of what is fixed or changed

Replaced deprecated np.matrix with np.array in physics module.

#### Release Notes

* physics.quantum
   * COMPATIBILITY BREAK: The deprecated `numpy.matrix` is replaced by `numpy.ndarray` (potentially breaking downstream code).

### CODE REVIEW

**Comment by oscargus** [comment]
And, yes, the idea was to add text in the compatibility break section.

**Comment by moorepants** [comment]
+1 it has been many years that numpy has deprecated this, so we should remove it by now.

**Comment by oscarbenjamin** [comment]
Okay, merging.

File: sympy/printing/tests/test_numpy.py [hunk 1]
```diff
@@ -83,9 +83,9 @@ def test_codegen_einsum():
     cg = convert_matrix_to_array(M * N)
     f = lambdify((M, N), cg, 'numpy')
 
-    ma = np.matrix([[1, 2], [3, 4]])
-    mb = np.matrix([[1,-2], [-1, 3]])
-    assert (f(ma, mb) == ma*mb).all()
+    ma = np.array([[1, 2], [3, 4]])
+    mb = np.array([[1,-2], [-1, 3]])
+    assert (f(ma, mb) == np.matmul(ma, mb)).all()
```
Comment (moorepants) [inline sympy/printing/tests/test_numpy.py:88]: `ma @ mb` works now too.

**Pull request merged**